sched/clock/clock_getcpuclockid.c

```diff
@@ -0,0 +1,74 @@
+/****************************************************************************
+ * sched/clock/clock_getcpuclockid.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <time.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include "clock/clock.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: clock_getcpuclockid
+ *
+ * Description:
+ *   The function shall return clock id of the CPU-time clock of the process
+ *   specified by pid
+ *
+ * Input Parameters:
+ *   pid - the specified process id
+ *   clockid - the clock type that need to setup
+ *
+ * Returned Value:
+ *   Return 0 on success, return error number on error
+ *
+ ****************************************************************************/
+
+int clock_getcpuclockid(pid_t pid, FAR clockid_t *clockid)
+{
+  if (pid < 0)
+    {
+      set_errno(EINVAL);
+      return ERROR;
+    }
```

Comment:
This in fact also is not compatible with https://pubs.opengroup.org/onlinepubs/9699919799/functions/clock_getcpuclockid.html
Most probably should be
```
  if (pid == 0)
    {
      pid = getpid();
    }
  else if (kill(pid, 0) < 0)
    {
      return ERROR;
    }
```

Reply:
Addressed here https://github.com/apache/nuttx/pull/9735